rest/stargate.go

```diff
@@ -0,0 +1,129 @@
+package rest
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/types/query"
+	"github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/gin-gonic/gin"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/likecoin/likecoin-chain-tx-indexer/db"
+	"github.com/likecoin/likecoin-chain-tx-indexer/logger"
+)
+
+func getOffset(query url.Values) (uint64, error) {
+	return getUint(query, "pagination.offset")
+}
+
+func isOrderByDesc(query url.Values) (bool, error) {
+	orderByStr := strings.ToUpper(query.Get("order_by"))
+	switch orderByStr {
+	case "", "ORDER_BY_UNSPECIFIED", "ORDER_BY_ASC":
+		return false, nil
+	case "ORDER_BY_DESC":
+		return true, nil
+	default:
+		return false, fmt.Errorf("available values for order_by: ORDER_BY_UNSPECIFIED, ORDER_BY_ASC, ORDER_BY_DESC")
+	}
+}
+
+func trimSingleQuotes(s string) (string, error) {
+	if len(s) >= 2 {
+		if c := s[len(s)-1]; s[0] == c && c == '\'' {
```

Comment:
Hmm although the code is correct, I find this hard to follow, since it is saying `if the first character is the same as the last character and the last character is '\'', then...`, and I think `if the first character is '\'' and the last character is '\'', then...` is more straight forward

Reply:
I see, since it's a remaining function modify from determining none/single/double quotes, I did forget to use a more straightforward determination.